tools/mirror/src/key_mapping.rs

```diff
@@ -90,18 +92,22 @@ pub fn map_key(key: &PublicKey, secret: Option<&[u8; crate::secret::SECRET_LEN]>
     }
 }
 
-// If it's an implicit account, interprets it as an ed25519 public key, maps that and then returns
-// the resulting implicit account. Otherwise does nothing. We do this so that transactions creating
-// an implicit account by sending money will generate an account that we can control
+// If it's a NEAR-implicit account, interprets it as an ed25519 public key,
+// maps that and then returns the resulting implicit account. Otherwise does nothing.
+// We do this so that transactions creating an implicit account
+// by sending money will generate an account that we can control.
 pub fn map_account(
     account_id: &AccountId,
     secret: Option<&[u8; crate::secret::SECRET_LEN]>,
 ) -> AccountId {
-    if account_id.is_implicit() {
-        let public_key = PublicKey::from_implicit_account(account_id).expect("must be implicit");
-        let mapped_key = map_key(&public_key, secret);
-        hex::encode(mapped_key.public_key().key_data()).parse().unwrap()
-    } else {
-        account_id.clone()
+    match account_id.get_account_type() {
+        AccountType::NearImplicitAccount => {
+            let public_key =
+                PublicKey::from_near_implicit_account(account_id).expect("must be implicit");
+            let mapped_key = map_key(&public_key, secret);
+            derive_near_implicit_account_id(&mapped_key.public_key())
+        }
+        AccountType::EthImplicitAccount => account_id.clone(),
```

Comment:
We'd likely want something similar to the near implicit for the eth implicit. I'm fine with either doing it here or adding a todo. 
@marcelo-gonzalez any thoughts on adding support for eth implicit accounts to mirror?

Reply:
LGTM this snippet here. I think it probably shouldn't be too hard to add explicit support for it here, but not super high priority at all

Reply:
The support will be added here in the next PR (that will treat ETH-implicit accounts and named accounts differently).